### PR TITLE
Making export

### PR DESCRIPTION
**Pull request by parteekcoder**
issue #366 

@NikhilM98 @llaske 
On clicking the export button which I added , based on filters it will download the result

### CODE REVIEW

**Comment by NikhilM98** [comment]
Don't you already have a PR open addressing this issue?

**Comment by parteekcoder** [comment]
> Don't you already have a PR open addressing this issue?

yeah in that PR you removed my confusion regarding the feature actually required , that's why I shifted to new

**Comment by parteekcoder** [comment]
Hey @llaske @NikhilM98 please review this PR

What I have done here is:-
when you click on `export assignment` button it will download the CSV files based on `filters`

I would to like hear your suggestions and work upon them

**Comment by NikhilM98** [comment]
@parteekcoder there are some linting issues with this PR. Can you handle them?

**Comment by parteekcoder** [comment]
> @parteekcoder there are some linting issues with this PR. Can you handle them?

sure I will remove that , but there is also an other PR having the same code 396

**Comment by parteekcoder** [comment]
> @parteekcoder there are some linting issues with this PR. Can you handle them?

yeah @NikhilM98 , now I removed all the lint errors thanks for reminding

Now you can review

**Comment by parteekcoder** [comment]
hey @NikhilM98 I removed all eslint error in my code

**Comment by NikhilM98** [comment]
> sure I will remove that , but there is also an other PR having the same code 396

It's okay to have another PR for the same issue if it's intended for GSoC. The quality of PRs is an important factor for evaluating the understanding of a contributor about the project.

**Comment by parteekcoder** [comment]
> > sure I will remove that , but there is also an other PR having the same code 396
> 
> It's okay to have another PR for the same issue if it's intended for GSoC. The quality of PRs is an important factor for evaluating the understanding of a contributor about the project.

@NikhilM98 If you would like anything needs to be improved in the PR , I will appreciate that and will update it

File: dashboard/views/deliveries.ejs [hunk 1]
```diff
@@ -28,6 +28,53 @@
                                 <div class="col-md-3"></div>
                                 <div class="col-md-6">
                                     <button class="btn btn-round pull-right" type="submit" data-l10n-id="showResults"><div class="ripple-container"></div></button>
+                                    <div class="btn pull-right btn-round" onclick="downloadAssignmentDeliveries()" data-l10n-id="ExportAssignemnt" id="users-exportassignment">Export Assignment</div>
```
Comment (NikhilM98) [inline dashboard/views/deliveries.ejs]: Looks like this text is not translated.

File: dashboard/controller/assignments/exportAssignment.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// include libraries
+var superagent = require('superagent'),
+	async = require('async'),
+	common = require('../../helper/common');
+
+module.exports = function exportAssignment(req, res) {
+	var resultAssignment = [];
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportAssignment.js]: Make this variable plural.

File: dashboard/controller/assignments/exportAssignment.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		if (assignmentCheck.content[0].metadata.activity) {
+			assign.activity = assignmentCheck.content[0].metadata.activity;
+		}
+		if (assignmentCheck.content[0].metadata.isSubmitted) {
+			assign.isSubmitted = assignmentCheck.content[0].metadata.isSubmitted;
+		}
+		return assign;
+	}
+	
+	async.series([
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportAssignment.js]: There was no reason to use .series()

**Review by NikhilM98 — CHANGES_REQUESTED** [review]

File: dashboard/controller/assignments/exportAssignment.js (hunk 2 above)
Comment (parteekcoder) [reply dashboard/controller/assignments/exportAssignment.js]: yeah sure @NikhilM98

File: dashboard/views/deliveries.ejs (hunk 1 above)
Comment (parteekcoder) [reply dashboard/views/deliveries.ejs]: yeah I will update this in `dashboard/public/I10n/locales.ini`

**Comment by parteekcoder** [comment]
@NikhilM98 done the changes you suggested , I can also update the fields which you would like to keep in `.csv` file 

Thanks for your review

**Comment by parteekcoder** [comment]
hey @llaske @NikhilM98 @ricknjacky please can you review this pull request so that it can be merged. And we can the discuss the fields to be exported in the csv file

**Comment by llaske** [comment]
Hi @parteekcoder ,
Thanks for this contribution.
Some information are missing: student name, delivery date and score.

image

**Comment by parteekcoder** [comment]
Yeah sure @llaske i will add those , thanks for testing and your invaluable time.

If you have any suggestions I will be happy to work on them

**Comment by parteekcoder** [comment]
Hi @llaske @NikhilM98 @ricknjacky ,
Thanks @llaske for testing this PR, I included the fields you told. And updated the PR

image

Please let me know if there are other changes to be made to improve this PR

File: dashboard/controller/assignments/exportAssignment.js [hunk 4]
```diff
@@ -46,6 +58,10 @@ module.exports = function exportAssignment(req, res) {
 		.query(q)
 		.end(function (error, response) {
 			if (response.statusCode == 200) {
+				// console.log(response.body.deliveries[0]);
+				// console.log(response.body.deliveries[0].content);
+				// console.log(response.body.deliveries[0].content[0]);
+				// console.log(response.body.deliveries[0].content[0].metadata);
```
Comment (llaske) [inline dashboard/controller/assignments/exportAssignment.js]: Please remove debug message.

**Review by llaske — CHANGES_REQUESTED** [review]
It's nice. Please fix my comment.

File: dashboard/controller/assignments/exportAssignment.js (hunk 4 above)
Comment (parteekcoder) [reply dashboard/controller/assignments/exportAssignment.js]: Oh!! yes , i forget to remove it. I will update the pr

**Comment by llaske** [comment]
Thanks @parteekcoder , good job.

**Comment by parteekcoder** [comment]
@llaske thanks for testing efforts and the suggestions you give to improve code quality

**Pull request merged**